core/server/common/src/main/java/alluxio/ReflectionUtils.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import org.slf4j.Logger;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.nio.charset.Charset;
+
+/**
+ * General reflection utils.
+ */
+public final class ReflectionUtils {
+  private static final ThreadMXBean THREAD_BEAN =
+      ManagementFactory.getThreadMXBean();
+
+  private static String getTaskName(long id, String name) {
+    if (name == null) {
+      return Long.toString(id);
+    }
+    return id + " (" + name + ")";
+  }
+
+  /**
+   * Print all of the thread's information and stack traces.
```

Comment:
nit, alluxio convention
```suggestion
   * Prints the information and stack traces of all threads.
```